src/swell/tasks/link_geos_output.py

```diff
@@ -27,73 +30,87 @@ def execute(self):
         Linking proper GEOS output files for JEDI to ingest and produce analysis.
         This will depend on the model type (ocean vs. atmosphere), model output
         type (history vs. restart), DA method, and window length.
-
-        TODO: Options could be offered in yamls, for instance history vs. restart.
         """
 
         self.current_cycle = os.path.basename(os.path.dirname(self.forecast_dir()))
 
-        # Create source and destination files for linking model output to SOCA
-        # ----------------------------------
-        src_dst_dict = {}
-
-        # src, dst = self.link_mom6_restart()
-        src, dst = self.link_mom6_history()
-        src_dst_dict[src] = dst
-
-        # Link CICE6 restart (iced.nc) and create SOCA input file (cice.res.nc)
-        # ---------------------------------------------------------------------
-        src, dst = self.prepare_cice6()
-        src_dst_dict[src] = dst
-
-        for src, dst in src_dst_dict.items():
-            if os.path.exists(src):
-                self.geos.linker(src, dst, self.cycle_dir())
+        # Parse configuration
+        # -------------------
+        marine_models = self.config.marine_models()
+        self.window_type = self.config.window_type()
+        self.window_length = self.config.window_length()
+        self.window_offset = self.config.window_offset()
+        self.window_begin_iso = self.da_window_params.window_begin_iso(self.window_offset)
+
+        if self.window_type == '4D' or 'fgat' in self.suite_name():
+            self.background_frequency = self.config.background_frequency()
+
+        self.bkgr_time_iso, self.bkgr_time_dto = self.da_window_params.local_background_time(
+            self.window_offset,
+            self.window_type,
+            dto=True)
+
+        # Create source and destination files for linking model output to cycle directories
+        # -----------------------------------------------------------------------------------
+        self.src_dst_dict: Dict[str, Tuple[str, ...]] = {}
+
+        if self.get_model() == 'geos_ocean' or self.get_model() == 'geos_marine':
```

Comment:
Stylistically, I have a slight preference for the following. It's shorter, and IMHO, the intent is clearer.

```suggestion
        if self.get_model() in ('geos_ocean', 'geos_marine'):
```

Reply:
Thanks, great suggestion!